scipy/signal/ltisys.py

```diff
@@ -1106,3 +1119,712 @@ def freqresp(system, w=None, n=10000):
+            # all vectors in ker_pole_j, this solves 100% of the problems and is
+            # still a valid choice for transfer_matrix. Indeed for complex poles
+            # we are sure to have a non zero imaginary part that way, and the
+            # problem of lines full of zeros in transfer_matrix is solved too as
+            # when a vector from ker_pole_j has a zero the other one(s)
+            # (when ker_pole_j.shape[1]>1) for sure won't have a zero there.
+            transfer_matrix_j = np.sum(ker_pole_j, axis=1)[:, np.newaxis]
+            transfer_matrix_j = (transfer_matrix_j /
+                                 np.linalg.norm(transfer_matrix_j))
+            if ~np.isreal(poles[j]):  # complex pole
+                transfer_matrix_j = np.hstack([np.real(transfer_matrix_j),
+                                            np.imag(transfer_matrix_j)])
+                ker_pole.extend([ker_pole_j, ker_pole_j])
+
+                #skip next pole as it is the conjugate
+                skip_conjugate = True
+            else:  # real pole nothing to do
+                ker_pole.append(ker_pole_j)
+
+            if j == 0:
+                transfer_matrix = transfer_matrix_j
+            else:
+                transfer_matrix = np.hstack((transfer_matrix, transfer_matrix_j))
+
+        if B.shape[1] > 1:  # otherwise there is nothing we can optimize
+            stop, cur_rtol, nb_iter = update_loop(ker_pole, transfer_matrix,
+                                                  poles, B, maxiter, rtol)
+            if not stop and rtol > 0:
+                # if rtol<=0 the user has probably done that on purpose,
+                # don't annoy him
+                err_msg = (
+                    "Convergence was not reached after maxiter iterations.\n"
+                    "You asked for a relative tolerance of %f we got %f" %
+                    (rtol, cur_rtol)
+                    )
+                warnings.warn(err_msg)
+
+        # reconstruct transfer_matrix to match complex conjugate pairs,
+        # ie transfer_matrix_j/transfer_matrix_j+1 are
+        # Re(Complex_pole), Im(Complex_pole) now and will be Re-Im/Re+Im after
+        transfer_matrix = transfer_matrix.astype(complex)
+        idx = 0
+        while idx < poles.shape[0]-1:
+            if ~np.isreal(poles[idx]):
+                rel = transfer_matrix[:, idx].copy()
+                img = transfer_matrix[:, idx+1]
+                # rel will be an array referencing a column of transfer_matrix
+                # if we don't copy() it will changer after the next line and
+                # and the line after will not yield the correct value
+                transfer_matrix[:, idx] = rel-1j*img
+                transfer_matrix[:, idx+1] = rel+1j*img
+                idx += 1  # skip next one
+            idx += 1
+
+        try:
+            m = np.linalg.solve(transfer_matrix.T, np.dot(np.diag(poles),
```

Comment:
I think for duplicate ``linalg`` functions we prefer to use ``numpy``, because eventually we want to remove the overlap by deprecating ``scipy.linalg`` functions. But given the large impact and that APIs of some functions have diverged, I don't expect that to happen any time soon. So in practice it doesn't matter much. And in this case we want to rely on a to-be-added feature of scipy's ``qr``, so it's fine to use ``scipy.linalg.qr`` I'd say.